File: Data Structures & Algorithms/find-pivot-index/submission-0.java
class Solution {
    public int pivotIndex(int[] nums) {
        int n = nums.length;
        int[] prefix = new int[n + 1];
        for(int i = 0; i < n; i++){
            prefix[i + 1] = prefix[i] + nums[i];
        }
        for(int i = 0; i < n; i++){
            if (prefix[i] == prefix[n] - prefix[i + 1]) return i;
        }
        return -1;
    }
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l = 0
        sum = 0
        length = 0
        minLen = float('inf')
        for r in range(len(nums)):
            sum += nums[r]
            length = r-l+1
            while sum >= target:
                minLen = min(length, minLen) 
                sum -= nums[l]
                l += 1
                length -= 1
        return minLen if minLen != float('inf') else 0